File: leetcode.com/1800_maximum_ascending_subarray_sum/solution.go
func Max(a, b int) int {
    if a > b {
        return a
    }

    return b
}

func maxAscendingSum(nums []int) int {
    total := nums[0]
    max := nums[0]

    for i := 0 ; i < len(nums) ; i ++ {
        if i > 0 && nums[i] > nums[i - 1] {
            total += nums[i]
        } else {
            max = Max(max, total)
            total = nums[i]
        }
    }

    return Max(max, total)
}

File: leetcode.com/1800_maximum_ascending_subarray_sum/solution.c
#define MAX(a, b)   (a > b ? a : b)

int maxAscendingSum(int *nums, int numsSize)
{
    int max = nums[0];
    int total = nums[0];

    for (int i = 0 ; i < numsSize ; i++) {
        if (i > 0 && nums[i] > nums[i - 1]) {
            total += nums[i];
        } else {
            max = MAX(max, total);
            total = nums[i];
        }
    }

    return MAX(max, total);
}
